ui/components/ui/currency-display/currency-display.component.js

```diff
@@ -27,25 +28,30 @@ export default function CurrencyDisplay({
     denomination,
     currency,
     suffix,
+    type,
   });
   return (
-    <div
-      className={classnames('currency-display-component', className)}
-      data-testid={dataTestId}
-      style={style}
-      title={(!hideTitle && title) || null}
-    >
-      {prefixComponent}
-      <span className="currency-display-component__text">
-        {parts.prefix}
-        {parts.value}
-      </span>
-      {parts.suffix && (
-        <span className="currency-display-component__suffix">
-          {parts.suffix}
-        </span>
-      )}
-    </div>
+    <>
+      {parts ? (
```

Comment:
when would parts not be defined?

Reply:
So part of what this PR does is in cases where we don't have a valid conversion rate we no longer show the fiat/user-preferred currency. the useCurrencyDisplay hook can return [null, null] if we can't provide an accurate value for a currency that requires conversion. So for instance in the eth-overview.js component if we don't have a conversionRate for a secondary currency it will return null and we won't show a secondary currency display. This pattern still leaves a lot to be desired but it was an intermediate step for a larger refactor that I still hope to complete soonish.